src-tauri/src/main.rs

```diff
@@ -7,24 +7,81 @@ mod mesh;
 mod state;
 
 use std::sync::Arc;
-use tauri::async_runtime;
+use tauri::{async_runtime, Manager};
+
+fn handle_cli_matches(
+    app: &mut tauri::App,
+    inital_autoconnect_state: &mut state::AutoConnectState,
+) -> Result<(), String> {
+    match app.get_cli_matches() {
```

Comment:
What happens if a user doesn't specify a port and has no default saved? Also, why are we saving the default port if we don't persist state across app launches?

Reply:
Fair point, the reason we're storing the autoconnect port in Tauri state is so that we can persist it across UI restarts. I also couldn't find an elegant way to immediately connect to a serial port in the `startup()` method (since it's not `async` :/). If they don't specify a port, the `port` argument will be passed with `{ occurrences: 0 }`, meaning the method will just ignore it and not set the state field.

Reply:
Gotcha, makes sense. thanks for the explanation!